lib/galaxy_test/api/test_history_contents.py

```diff
@@ -737,6 +737,38 @@ def test_index_filter_by_name_ignores_case(self, history_id):
         ).json()
         assert len(contents_response) == 0
 
+    @skip_without_tool("cat_data_and_sleep")
+    def test_index_filter_by_related_items(self, history_id):
+        # initialise history with 2 datasets
+        input_hda_id = self.dataset_populator.new_dataset(history_id)["id"]
+        unrelated_hid = self.dataset_populator.new_dataset(history_id)["hid"]
+
+        # Run tool on first dataset to get 3rd, related dataset
+        inputs = {
+            "input1": {"src": "hda", "id": input_hda_id},
+            "sleep_time": 10,
```

Comment:
```suggestion
            "sleep_time": 0,
```
I don't think there's a reason to slow down the test ?

Reply:
I can try that, I saw a sleep being  used for that tool everywhere else. But good point, I’ll try without it.

Reply:
Done, set the `sleep_time: 0`